x-pack/packages/kbn-elastic-assistant/impl/assistant/index.tsx

```diff
@@ -260,90 +241,10 @@ const AssistantComponent: React.FC<Props> = ({
     [allSystemPrompts, conversations]
   );
 
-  const handlePromptChange = useCallback((prompt: string) => {
-    setPromptTextPreview(prompt);
-    setUserPrompt(prompt);
-  }, []);
-
-  // Handles sending latest user prompt to API
-  const handleSendMessage = useCallback(
-    async (promptText) => {
-      const onNewReplacements = (newReplacements: Record<string, string>) =>
-        appendReplacements({
-          conversationId: selectedConversationId,
-          replacements: newReplacements,
-        });
-
-      const systemPrompt = allSystemPrompts.find((prompt) => prompt.id === editingSystemPromptId);
-
-      const message = await getCombinedMessage({
-        isNewChat: currentConversation.messages.length === 0,
-        currentReplacements: currentConversation.replacements,
-        onNewReplacements,
-        promptText,
-        selectedPromptContexts,
-        selectedSystemPrompt: systemPrompt,
-      });
-
-      const updatedMessages = appendMessage({
-        conversationId: selectedConversationId,
-        message,
-      });
-
-      // Reset prompt context selection and preview before sending:
-      setSelectedPromptContexts({});
-      setPromptTextPreview('');
-
-      const rawResponse = await sendMessages({
-        http,
-        apiConfig: currentConversation.apiConfig,
-        messages: updatedMessages,
-      });
-      const responseMessage: Message = getMessageFromRawResponse(rawResponse);
-      appendMessage({ conversationId: selectedConversationId, message: responseMessage });
-    },
-    [
-      allSystemPrompts,
-      currentConversation.messages.length,
-      currentConversation.replacements,
-      currentConversation.apiConfig,
-      selectedPromptContexts,
-      appendMessage,
-      selectedConversationId,
-      sendMessages,
-      http,
-      appendReplacements,
-      editingSystemPromptId,
-    ]
-  );
-
-  const handleButtonSendMessage = useCallback(() => {
-    handleSendMessage(promptTextAreaRef.current?.value?.trim() ?? '');
-    setUserPrompt('');
-  }, [handleSendMessage, promptTextAreaRef]);
-
   const handleOnSystemPromptSelectionChange = useCallback((systemPromptId?: string) => {
     setEditingSystemPromptId(systemPromptId);
   }, []);
 
-  const handleOnChatCleared = useCallback(() => {
-    const defaultSystemPromptId = getDefaultSystemPrompt({
-      allSystemPrompts,
-      conversation: conversations[selectedConversationId],
-    })?.id;
-
-    setPromptTextPreview('');
-    setUserPrompt('');
-    setSelectedPromptContexts({});
-    clearConversation(selectedConversationId);
-    setEditingSystemPromptId(defaultSystemPromptId);
-  }, [allSystemPrompts, clearConversation, conversations, selectedConversationId]);
-
-  const shouldDisableConversationSelectorHotkeys = useCallback(() => {
```

Comment:
@andrew-goldstein @spong this was the only dependency between the chat prompt and the conversation selector. I actually could not get the hot key behavior on main, so I just removed this... maybe i should not have?

Reply:
@spong could this be related to the observation that navigation between multiple conversations via the shortcut keys occasionally got "stuck"? That issue doesn't appear to be reproducible in this branch.

Reply:
This is what enabled the disabling of the conversation quick switch shortcut key when the user prompt input had focus (which might've been what you were seeing?), as to not prevent on the text navigation shortcut that is shared w/ the conversation quick switch. 

If keeping the conversation switching shortcut (and I think we should), we should just update the shortcut key so we can avoid this outright. I'll test on `main` and open an issue or push a fix if something is awry here though 🙂